include/RAJA/exec-cuda/MemUtils_CUDA.hxx

```diff
@@ -62,37 +62,90 @@
 
 namespace RAJA
 {
+/// Maximum number of blocks that RAJA will launch
+/// threads are reused to finish all work
+#define RAJA_CUDA_MAX_NUM_BLOCKS (1024 * 16)
 
 /// Size of reduction memory block for each reducer object (value based on
 /// rough estimate of "worst case" -- need to think more about this...
-#define RAJA_CUDA_REDUCE_BLOCK_LENGTH (1024 + 8) * 16
+#define RAJA_CUDA_REDUCE_BLOCK_LENGTH RAJA_CUDA_MAX_NUM_BLOCKS
 
 /// Reduction Tallies are computed into a small block to minimize UM migration
 #define RAJA_CUDA_REDUCE_TALLY_LENGTH RAJA_MAX_REDUCE_VARS
 
-///
-/// Typedef defining common data type for RAJA-Cuda reduction data blocks
-/// (use this in all cases to avoid type confusion).
-///
-typedef double CudaReductionBlockDataType;
+/// Should be large enough for all types for which cuda atomics exist
+/// includes the size of the index variable for Loc reductions
+#define RAJA_CUDA_REDUCE_VAR_MAXSIZE 16
```

Comment:
This is used to setup the dummy type to handle up to the worst case reduction variable size (eg minmaxloc)

and is used a few lines later in the following
/// dummy types for use in allocating arrays and distributing array segments                                                                                                                                                                                                                             
struct CudaReductionDummyDataType {                                                                                                                                                                                                                                                                      
  unsigned char data[RAJA_CUDA_REDUCE_VAR_MAXSIZE];                                                                                                                                                                                                                                                      
};